cmd/nerdctl/main.go

```diff
@@ -71,10 +71,22 @@ func main() {
 }
 
 func xmain() error {
-	if len(os.Args) == 3 && os.Args[1] == logging.MagicArgv1 {
+	loggingMode := false
+	for _, arg := range os.Args {
+		if arg == logging.MagicArgv1 {
+			loggingMode = true
+			break
+		}
+	}
```

Comment:
Sorry for going back and forth, but this logic doesn’t seem robust.
We have to ensure that MagicArgv1 is always argv1.
Other args should be encoded in argv2 or in the metadata referred by argv2.